include/tscore/ts_file.h

```diff
@@ -0,0 +1,245 @@
+/** @file
+
+  Simple path and file utilities.
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#pragma once
+
+#include <string>
+#include <string_view>
+#include <array>
+#include <system_error>
+#include <sys/stat.h>
+#include "tscore/ink_memory.h"
+#include "tscpp/util/TextView.h"
+#include "tscore/BufferWriter.h"
+
+namespace ts
+{
+namespace file
+{
+  /** Utility class for file system paths.
+   */
+  class path
+  {
+    using self_type = path;
+
+  public:
+    static constexpr char SEPARATOR = '/';
```

Comment:
should be or add

typedef char value_type;
typedef basic_string<value_type> string_type;
static constexpr value_type preferred_separator = '/';

SEPARATOR should be removed or we should avoid usage of it outside this class/function set. Maybe change it to _SEPARATOR or use preferred_separator.